app/models/salt_handler/minion_highstate.rb

```diff
@@ -29,9 +29,10 @@ def process_event
     data = JSON.parse(salt_event.data)
 
     highstate_succeeded = data["success"] && data["retcode"].zero?
-    new_highstate = Minion.highstates[highstate_succeeded ? :applied : :failed]
     # rubocop:disable SkipsModelValidations
-    minion.update_column(:highstate, new_highstate)
+    minion.update_column(:highstate, Minion.highstates[:failed]) unless highstate_succeeded
```

Comment:
i cant make the connection yet, if `highstate_succeeded` is `true` here, don't we need to `minion.update_column(:highstate, :applied)` here? what if the last `highstate` is still `:failed` ?

Reply:
The highstate will only mark failure if the highstate failed. The responsible for marking all pending nodes as succeeded (applied) will be the orchestration result. Since we are lacking a proper overall orchestration result in the database structure, this is a way to make sure that nodes are marked as "applied" when the orchestration has finished and has finished successfully.